apps/web/components/SideNav.tsx

```diff
@@ -190,6 +196,61 @@ export function SideNav(props: {
 
 function Sidebar(props: { isMobile: boolean }) {
   const path = usePathname();
+  const { data, isLoading, error } =
+    useSWR<LabelsResponse>("/api/google/labels");
+
+  const [open, setOpen] = useState<boolean>();
+
+  function tagItem(list: LabelsResponse["labels"]) {
+    return list?.map((label) => (
+      <a
+        href="#"
+        className={clsx(
+          "group flex gap-x-3 rounded-md p-2 text-sm font-semibold leading-6 text-gray-400 hover:bg-gray-800 hover:text-white",
+        )}
+        key={label.id}
+      >
+        <TagIcon className="h-4 w-4 shrink-0" aria-hidden="true" />
+        {label.name}
+      </a>
+    ));
+  }
+
+  const LabelSection = (
+    <>{tagItem(data?.labels?.filter((label) => label?.type === "user"))}</>
+  );
+
+  const LabelSectionExpandable = (
```

Comment:
This is rendered on the UI, if the user has more than 4 "user" created labels in their account as an expandable section, similar to GMail's UI.

Reply:
https://github.com/elie222/inbox-zero/pull/97#issuecomment-1923793958